### PR TITLE
Add addSubscriptionFilter to CloudwatchService

### LINKED ISSUE #62: Auto Kinesis setup part 5: Create the subscription
Programmatically create the CloudWatch log subscription using the policy created in #124. See these docs for reference.

https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/Subscriptions.html
https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/CreateSubscriptionFilter.html

### PR DESCRIPTION
**Pull request by ceruleancee**
Closes #62

### CODE REVIEW

File: src/main/java/org/graylog/integrations/aws/service/CloudWatchService.java [hunk 1]
```diff
@@ -54,4 +60,32 @@ public LogGroupsResponse getLogGroupNames(String region, String awsAccessKeyId,
...
+            final String responseMessage = String.format("Success. The subscription filter [%s] was added to [%s].",
+                                                         filterName, logGroup);
+            return responseMessage;
+        } catch (Exception e) {
+            final String specificError = ExceptionUtils.formatMessageCause(e);
+            final String responseMessage = String.format("Attempt to add subscription [%s] to Cloudwatch log group " +
+                                                         "[%s] failed due to the following exception: [%s]",
+                                                         filterName, logGroup, specificError);
+            LOG.error(responseMessage);
+            throw new InternalServerErrorException(responseMessage, e);
```
Comment (danotorrey) [inline src/main/java/org/graylog/integrations/aws/service/CloudWatchService.java]: What do you think about throwing `javax.ws.rs.BadRequestException` here instead? `BadRequestException` will respond with a 400 HTTP code indicating that something is wrong with the request parameters. 

`InternalServerErrorException` returns a 500 HTTP code which generally says that it was the Graylog servers fault. 

I think the `BadRequestException` fits better, but happy to discuss more too. 

More info on statuses: https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.4.1

File: src/main/java/org/graylog/integrations/aws/service/KinesisService.java [hunk 2]
```diff
@@ -423,4 +423,5 @@ public KinesisNewStreamResponse createNewKinesisStream(KinesisNewStreamRequest k
             throw new InternalServerErrorException(responseMessage, e);
         }
     }
+
```
Comment (danotorrey) [inline src/main/java/org/graylog/integrations/aws/service/KinesisService.java]: Can this extra line be removed?

File: src/main/java/org/graylog/integrations/aws/service/CloudWatchService.java [hunk 3]
```diff
@@ -54,4 +60,32 @@ public LogGroupsResponse getLogGroupNames(String region, String awsAccessKeyId,
...
+        PutSubscriptionFilterRequest putSubscriptionFilterRequest =
+                PutSubscriptionFilterRequest.builder()
+                                            .logGroupName(logGroup)
+                                            .filterName(filterName)
+                                            .filterPattern(filterPattern)
+                                            .destinationArn(streamArn)
+                                            .roleArn(roleArn)
+                                            .distribution(Distribution.BY_LOG_STREAM)
+                                            .build();
+        sleep(2_000);
```
Comment (danotorrey) [inline src/main/java/org/graylog/integrations/aws/service/CloudWatchService.java]: Is this sleep statement required?

File: src/main/java/org/graylog/integrations/aws/service/CloudWatchService.java [hunk 4]
```diff
@@ -54,4 +60,32 @@ public LogGroupsResponse getLogGroupNames(String region, String awsAccessKeyId,
...
+        try {
+            cloudWatch.putSubscriptionFilter(putSubscriptionFilterRequest);
+            final String responseMessage = String.format("Success. The subscription filter [%s] was added to [%s].",
+                                                         filterName, logGroup);
+            return responseMessage;
+        } catch (Exception e) {
+            final String specificError = ExceptionUtils.formatMessageCause(e);
+            final String responseMessage = String.format("Attempt to add subscription [%s] to Cloudwatch log group " +
```
Comment (danotorrey) [inline src/main/java/org/graylog/integrations/aws/service/CloudWatchService.java]: This `responseMessage` variable is not needed and can be in inline return statement like this:
```suggestion
            return String.format("Attempt to add subscription [%s] to Cloudwatch log group " +
```

File: src/main/java/org/graylog/integrations/aws/service/CloudWatchService.java [hunk 5]
```diff
@@ -54,4 +60,32 @@ public LogGroupsResponse getLogGroupNames(String region, String awsAccessKeyId,
         LOG.debug("Log groups queried: [{}]", groupNameList);
         return LogGroupsResponse.create(groupNameList, groupNameList.size());
     }
+
+    public String addSubscriptionFilter(CloudWatchLogsClient cloudWatch, String logGroup, String streamArn,
+                                        String roleArn, String filterName,
+                                        String filterPattern) throws InterruptedException {
+        PutSubscriptionFilterRequest putSubscriptionFilterRequest =
```
Comment (danotorrey) [inline src/main/java/org/graylog/integrations/aws/service/CloudWatchService.java]: The `putSubscriptionFilterRequest` variable can be finalized. 
```suggestion
        final PutSubscriptionFilterRequest putSubscriptionFilterRequest =
```

File: src/main/java/org/graylog/integrations/aws/service/CloudWatchService.java (hunk 1 above)
Comment (ceruleancee) [reply src/main/java/org/graylog/integrations/aws/service/CloudWatchService.java]: Good thinking. That exception makes more sense.

File: src/main/java/org/graylog/integrations/aws/service/CloudWatchService.java (hunk 4 above)
Comment (ceruleancee) [reply src/main/java/org/graylog/integrations/aws/service/CloudWatchService.java]: I kept the responseMessage and instead included `throw new BadRequestException(responseMessage, e);`

File: src/main/java/org/graylog/integrations/aws/service/CloudWatchService.java (hunk 3 above)
Comment (ceruleancee) [reply src/main/java/org/graylog/integrations/aws/service/CloudWatchService.java]: no its not. good catch! deleting.

**Pull request merged**